libs/parallelism/algorithms/include/hpx/parallel/datapar/iterator_helpers.hpp

```diff
@@ -33,7 +33,7 @@ namespace hpx { namespace parallel { namespace util { namespace detail {
             typedef typename std::iterator_traits<Iter>::value_type value_type;
             return (reinterpret_cast<std::uintptr_t>(std::addressof(*it)) &
                        (traits::vector_pack_alignment<value_type>::value -
-                           1)) != 0;
+                           1)) == 0;
```

Comment:
Well spotted!

Reply:
I still need to fix this https://github.com/STEllAR-GROUP/hpx/blob/fixing_datapar/libs/parallelism/algorithms/include/hpx/parallel/datapar/zip_iterator.hpp#L35-L45 as https://app.circleci.com/pipelines/github/srinivasyadav18/hpx/40/workflows/f86b0f5f-1086-4d1a-b716-fc630649b721/jobs/613/tests#failed-test-1 this is failing because now is_data_aligned changed.